internal/app/retriever.go

```diff
@@ -0,0 +1,121 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/filecoin-project/lassie/pkg/lassie"
+	"github.com/filecoin-project/lassie/pkg/storage"
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/storage/deferred"
+	trustlessutils "github.com/ipld/go-trustless-utils"
+)
+
+// Retriever is responsible for retrieving file from the network.
+type Retriever struct {
+	cacheStore *cacheStore
+	coldStore  *coldStore // nolint
+}
+
+// NewRetriever creates a new Retriever.
+func NewRetriever(provider VaultsProvider) *Retriever {
+	return &Retriever{
+		cacheStore: &cacheStore{
+			provider: provider,
+		},
+	}
+}
+
+// Retrieve retrieves file from the network.
+func (r *Retriever) Retrieve(ctx context.Context, c cid.Cid, output string, name string) error {
+	if output == "-" {
+		return r.cacheStore.retrieveStdout(ctx, c)
+	}
+
+	return r.cacheStore.retrieveFile(ctx, c, output, name)
+}
+
+type cacheStore struct {
+	provider VaultsProvider
+}
+
+func (cs *cacheStore) retrieveStdout(ctx context.Context, cid cid.Cid) error {
+	if err := cs.provider.RetrieveEvent(ctx, RetrieveEventParams{
+		CID: cid,
+	}, os.Stdout); err != nil {
+		return fmt.Errorf("failed to retrieve to file: %s", err)
+	}
+
+	return nil
+}
+
+func (cs *cacheStore) retrieveFile(ctx context.Context, cid cid.Cid, output string, name string) error {
+	// Write to the provided path or current directory
+	if output == "" {
```

Comment:
Looks good. But curious why not use output == "" for "stdout" instead of hyphen and output == "." for current directory?

Reply:
It's because "-" as stdout is a convention commonly used in command-line programs